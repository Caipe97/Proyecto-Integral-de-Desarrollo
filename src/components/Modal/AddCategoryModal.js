import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button } from 'react-bootstrap';
import TextField from '@material-ui/core/TextField';
import Add from '../../images/add.png';

const AddCategoryModal = (props) =>{
  const [modalShow, setModalShow] = useState(false);
  const [state, setState] = useState({
    name: '',
    errorMessage: ''
  })

  const handleChange = event => {
    setState({
      ...state,
      [event.target.name]: event.target.value
    })
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    await props.onCreateCategory(state.name, props.userId);
    setState({
      name: '',
      errorMessage: ''
    });
  };

  return (
    <>
      {/* <Button  type="button" variant="primary" onClick={() => setModalShow(true)}>
        Crear Categoría
      </Button> */}
      <img src={Add} alt='agregar' style={{width: '30px', height: '30px', cursor: 'pointer'}} onClick={() => setModalShow(true)}/>
      <Modal
        show={modalShow}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
          {'Crear Categoría'}
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <form style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}>
            <TextField label="Nombre" name='name' type='name' value={state.name} onChange={handleChange} required />
            <button onClick={handleSubmit} className='button'>Crear</button>
            <p>{state.errorMessage}</p>
          </form>
        </Modal.Body>
        <Modal.Footer>
          <Button type="button" onClick={() => setModalShow(false)}>Cerrar</Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}
  
  export default AddCategoryModal;