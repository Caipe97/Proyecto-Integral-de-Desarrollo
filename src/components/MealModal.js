import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button } from 'react-bootstrap';

const MealModal =(props) =>{
  const [modalShow, setModalShow] = useState(false);
  let totalCaloriesPerFood = 0;
  const calculateTotalCaloriesPerMeal = () => {
    let totalCaloriesPerMeal = 0;
    for (let i = 0; i < props.meal.FoodList.length; i++) {
      const element = props.meal.FoodList[i];
      totalCaloriesPerMeal += element.food.caloriesPerServing * element.quantity;
    }
    return totalCaloriesPerMeal;
  }
  return (
    <>
      <Button  type="button" variant="primary" onClick={() => setModalShow(true)}>
        Mas info
      </Button>
      
      <Modal
        show={modalShow}
        //onHide={() => setModalShow(false)}
        // meal={props.meal}
        // history={props.history}
        // onDeleteMeal={props.onDeleteMeal}
        // onUpdateCurrentMealInState={props.onUpdateCurrentMealInState}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
          {props.meal.name}
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <h4>{props.meal.dateEaten.toString().replace('T',' ').substring(0,16)}</h4>
          <p>
            Alimentos consumidos:
          </p>
          {props.meal.FoodList.map((FoodListItem) => {
              totalCaloriesPerFood = FoodListItem.food.caloriesPerServing * FoodListItem.quantity
              return(
                <div key={FoodListItem.food.foodId} style={{textAlign:'center',top:'3px',background:'#B6E052'}}>
                  <p>{FoodListItem.quantity} x {FoodListItem.food.name}: {totalCaloriesPerFood} calorias en total</p>
                </div>
              );
            })
          }
          <p>Total de calorias consumidas en la comida: {calculateTotalCaloriesPerMeal()}</p>
        </Modal.Body>
        <Modal.Footer>
          <Button type="button" onClick={() => setModalShow(false)}>Cerrar</Button>
          <Button type="button" onClick={() =>  {props.onUpdateCurrentMealInState(props.meal); props.history.push("/meals", {meal: props.meal})}}>Editar</Button>
          <Button type="button" onClick={() =>  props.onDeleteMeal(props.meal.mealId, props.meal.userId)}>Eliminar</Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}
  
  export default MealModal;