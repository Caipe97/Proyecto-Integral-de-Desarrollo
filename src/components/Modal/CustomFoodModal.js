import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button } from 'react-bootstrap';
import TextField from '@material-ui/core/TextField';
import Combobox from "react-widgets/Combobox";
import "react-widgets/styles.css";
import Edit from '../../images/edit.png';

const CustomFoodModal = (props) =>{
  const [modalShow, setModalShow] = useState(false);
  const [state, setState] = useState({
    name: '',
    recommendedServing: '',
    caloriesPerServing: '',
    foodCategoryId: '',
    message: ''
  })

  const handleChange = event => {
    setState({
      ...state,
      [event.target.name]: event.target.value
    })
  };

  const handleChangeComboBox = event => {
    setState({
      ...state,
      foodCategoryId: event.foodCategoryId
    })
  };

  if(props.edit){
    
    const handleSubmitEdit = async (event) => {
      event.preventDefault();
      const data = await props.onEditCustomFood(props.foodId, state.name, state.recommendedServing, state.caloriesPerServing, state.foodCategoryId);
      if(data){
        setState({
          ...state,
          message: 'Alimento editado exitosamente'
        });
      }
    };

    return (
      <>
      <img src={Edit} alt='editar' style={{width: '20px', height: '20px', cursor: 'pointer'}} onClick={() => {
        setState({
          name: props.row.name,
          recommendedServing: props.row.recommendedServing,
          caloriesPerServing: props.row.caloriesPerServing,
          foodCategoryId: props.row.foodCategoryId
        });
        setModalShow(true)
        }}/>
        
        <Modal
          show={modalShow}
          size="lg"
          aria-labelledby="contained-modal-title-vcenter"
          centered
        >
          <Modal.Header>
            <Modal.Title id="contained-modal-title-vcenter">
            {'Editar Alimento Personalizado'}
            </Modal.Title>
            <Button type="button" onClick={() => {setModalShow(false); setState({...state, message: ''});}} style={{ backgroundColor: 'white', borderColor: 'white', color: "black"}}>X</Button>
          </Modal.Header>
          <Modal.Body>
            <form style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}>
              <TextField label="Nombre" name='name' type='name' value={state.name} onChange={handleChange} required />
              <TextField label="Porción Recomendada" name='recommendedServing' type='recommendedServing' value={state.recommendedServing} onChange={handleChange} required />
              <TextField label="Calorias por Porción" name='caloriesPerServing' type='caloriesPerServing' value={state.caloriesPerServing} onChange={handleChange} required />
              <Combobox
                data={props.foodCategories}
                textField='name'
                onSelect={handleChangeComboBox}
                groupBy={category => category.userId}
                renderListGroup={ ({group}) => ( //group es el userId
                  <span>{group === null ? 'Default' : 'Custom'}</span>
                )}
              />
              <button onClick={handleSubmitEdit} className='button'>Finalizar edición</button>
              <p>{state.message}</p>
            </form>
          </Modal.Body>
        </Modal>
      </>
    );
  } else {
    const handleSubmitAdd = async (event) => {
      event.preventDefault();
      const data = await props.onAddCustomFood(state.name, state.recommendedServing, state.caloriesPerServing, state.foodCategoryId, props.userId);
      if(data){
        setState({
          name: '',
          recommendedServing: '',
          caloriesPerServing: '',
          foodCategoryId: '',
          message: 'Alimento agregado exitosamente'
        });
      }
    };

    return (
      <>
        <Button  type="button" variant="primary" onClick={() => setModalShow(true)} style={{backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}}>
          Agregar Alimento Personalizado
        </Button>
        
        <Modal
          show={modalShow}
          size="lg"
          aria-labelledby="contained-modal-title-vcenter"
          centered
        >
          <Modal.Header>
            <Modal.Title id="contained-modal-title-vcenter">
            {'Agregar Alimento Personalizado'}
            </Modal.Title>
            <Button type="button" onClick={() => setModalShow(false)} style={{ backgroundColor: 'white', borderColor: 'white', color: "black"}}>X</Button>
          </Modal.Header>
          <Modal.Body>
            <form style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}>
              <TextField label="Nombre" name='name' type='name' value={state.name} onChange={handleChange} required />
              <TextField label="Porción Recomendada" name='recommendedServing' type='recommendedServing' value={state.recommendedServing} onChange={handleChange} required />
              <TextField label="Calorias por Porción" name='caloriesPerServing' type='caloriesPerServing' value={state.caloriesPerServing} onChange={handleChange} required />
              <Combobox
                data={props.foodCategories}
                textField='name'
                onSelect={handleChangeComboBox}
                groupBy={category => category.userId}
                renderListGroup={ ({group}) => ( //group es el userId
                  <span>{group === null ? 'Default' : 'Custom'}</span>
                )}
              />
              <button onClick={handleSubmitAdd} className='button'>Agregar</button>
              <p>{state.message}</p>
            </form>

          </Modal.Body>
        </Modal>
      </>
    );
  }

}
  
  export default CustomFoodModal;