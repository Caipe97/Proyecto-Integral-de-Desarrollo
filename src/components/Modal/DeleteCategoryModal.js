import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button } from 'react-bootstrap';
import Combobox from "react-widgets/Combobox";
import "react-widgets/styles.css";
import Delete from '../../images/delete.png';

const DeleteCategoryModal = (props) =>{
  const [modalShow, setModalShow] = useState(false);
  let onlyCustomCategories = props.foodCategories.filter((category) => category.userId !== null);

  const [state, setState] = useState({
    name: '',
    foodCategoryId: '',
    categoryName: '',
    message: ''
  })

  const handleSubmit = async (event) => {
    event.preventDefault();
    if(state.foodCategoryId !== ''){
      const data = await props.onDeleteCategory(state.foodCategoryId);
      if(data){
        setState({
          name: '',
          foodCategoryId: '',
          categoryName: '',
          message: 'Categoría eliminada exitosamente'
        });
      }
    } else {
      setState({...state, message: 'Debe seleccionar alguna categoría para eliminarla'});
    }
  };

  const getFoodCategoryName = (foodCategoryId) => {
    const foodCategoryName = props.foodCategories.find(foodCategory => foodCategory.foodCategoryId === foodCategoryId);
    if(foodCategoryId === -2){
      return '';
    }
    return foodCategoryName;
  }

  const onChangeComboBox = async event => {
    await setState({...state, foodCategoryId: event.foodCategoryId, categoryName: getFoodCategoryName(event.foodCategoryId)});
  }

  return (
    <>
      <img src={Delete} alt='eliminar' style={{width: '30px', height: '30px', cursor: 'pointer'}} onClick={() => setModalShow(true)}/>
      <Modal
        show={modalShow}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
          {'Eliminar Categoría'}
          </Modal.Title>
          <Button type="button" onClick={() => {setModalShow(false); setState({...state, message: ''});}} style={{ backgroundColor: 'white', borderColor: 'white', color: "black"}}>X</Button>
        </Modal.Header>
        <Modal.Body>

          <form style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}>
            <Combobox
              data={onlyCustomCategories}
              value={state.categoryName}
              textField='name'
              onSelect={onChangeComboBox}
              groupBy={category => category.userId}
              renderListGroup={ ({group}) => ( //group es el userId
                <span>{group === null ? 'Default' : 'Custom'}</span>
              )}
            />
            <button onClick={handleSubmit} className='button'>Eliminar</button>
            <p>{state.message}</p>
            {props.isPending
              ? <div className="spinner-border" role="status">
                  <span className="sr-only"></span>
                </div>
              : null}
          </form>
        </Modal.Body>
      </Modal>
    </>
  );
}
  
  export default DeleteCategoryModal;