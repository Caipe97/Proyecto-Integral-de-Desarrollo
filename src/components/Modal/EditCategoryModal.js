import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button } from 'react-bootstrap';
import Combobox from "react-widgets/Combobox";
import "react-widgets/styles.css";
import TextField from '@material-ui/core/TextField';
import Edit from '../../images/edit.png';

const EditCategoryModal = (props) =>{
  const [modalShow, setModalShow] = useState(false);
  let onlyCustomCategories = props.foodCategories.filter((category) => category.userId !== null);
  const [state, setState] = useState({
    name: '',
    foodCategoryId: '',
    categoryName: '',
    message: ''
  })

  const handleChange = event => {
    setState({
      ...state,
      [event.target.name]: event.target.value
    })
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    if(state.name !== '' && state.foodCategoryId !== ''){
      const data = await props.onEditCategory(state.foodCategoryId, state.name);
      if(data){
        setState({
          name: '',
          foodCategoryId: '',
          message: 'Categoría editada exitosamente'
        });
      }
    } else {
      setState({...state, message: 'Debe seleccionar alguna categoría y elegir un nuevo nombre'});
    }
  };

  const getFoodCategoryName = (foodCategoryId) => {
    const foodCategoryName = props.foodCategories.find(foodCategory => foodCategory.foodCategoryId === foodCategoryId);
    if(foodCategoryId === -2){
      return '';
    }
    return foodCategoryName;
  }

  const onChangeComboBox = async event => {
    await setState({...state, foodCategoryId: event.foodCategoryId, categoryName: getFoodCategoryName(event.foodCategoryId)});
  }

  return (
    <>
      <img src={Edit} alt='editar' style={{width: '30px', height: '30px', cursor: 'pointer'}} onClick={() => setModalShow(true)}/>
      <Modal
        show={modalShow}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
          {'Editar Categoría'}
          </Modal.Title>
          <Button type="button" onClick={() => {setModalShow(false); setState({...state, message: ''});}} style={{ backgroundColor: 'white', borderColor: 'white', color: "black"}}>X</Button>
        </Modal.Header>
        <Modal.Body>

          <form style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}>
            <Combobox
              data={onlyCustomCategories}
              value={state.categoryName}
              textField='name'
              onSelect={onChangeComboBox}
              groupBy={category => category.userId}
              renderListGroup={ ({group}) => ( //group es el userId
                <span>{group === null ? 'Default' : 'Custom'}</span>
              )}
            />
            <TextField label="Nuevo nombre" name='name' type='name' value={state.name} onChange={handleChange} required />
            <button onClick={handleSubmit} className='button'>Editar</button>
            <p>{state.message}</p>
            {props.isPending
              ? <div className="spinner-border" role="status">
                  <span className="sr-only"></span>
                </div>
              : null}
          </form>
        </Modal.Body>
      </Modal>
    </>
  );
}
  
  export default EditCategoryModal;