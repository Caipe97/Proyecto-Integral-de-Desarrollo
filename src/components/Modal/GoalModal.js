import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button, ProgressBar } from 'react-bootstrap';

const GoalModal =(props) =>{
  const [modalShow, setModalShow] = useState(false);

  const calculateCurrentCaloriesOfGoal = goal => {
    let currentCaloriesPerObjective = 0;
    for (let i = 0; i < goal.objectives.length; i++) {
      const objective = goal.objectives[i];
      currentCaloriesPerObjective += objective.currentCalories;
    }
    return currentCaloriesPerObjective;
  }

  const isGoalEditable = goal => {
    let dateStart = new Date(goal.dateStart);
    dateStart = new Date(dateStart.setHours(dateStart.getHours() + 3));
    const today = new Date();
    if(dateStart.getFullYear() > today.getFullYear()){
      return true;
    } else if(dateStart.getFullYear() === today.getFullYear()){
      if(dateStart.getMonth() > today.getMonth()){
        return true;
      } else {
        return false;
      }
    } else {
      return false;
    }
  }

  return (
    <>
      <Button  type="button" variant="primary" style={{backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}} onClick={() => setModalShow(true)}>
        Mas info
      </Button>
      
      <Modal
        show={modalShow}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
          {props.goal.name} 
          </Modal.Title>
          <Button type="button" onClick={() => setModalShow(false)} style={{ backgroundColor: 'white', borderColor: 'white', color: "black"}}>X</Button>

        </Modal.Header>
        <Modal.Body>
          <h4>{props.goal.dateStart.toString().replace('T',' ').substring(0,7)}</h4>
          <p>
            Objetivos:
          </p>
          
          {props.goal.objectives.map(objective => {
            return(
              <div key={objective.foodCategoryId} style={{textAlign:'center',top:'3px',background:'#B6E052', margin: "5px", padding: 5, borderRadius: 10}}>
                <p style={{margin: 0}}>{objective.foodCategory.name}: {objective.currentCalories} de {objective.objectiveCalories}</p>
                <ProgressBar animated 
                  variant={(objective.currentCalories/objective.objectiveCalories)*100 > 65 ? "danger" : "success"}  
                  now={(objective.currentCalories/objective.objectiveCalories)*100} 
                  label={`${(objective.currentCalories/objective.objectiveCalories)*100}%`}
                />
                  
              </div>
            );
            })
          }
          {
          props.goal.totalCalories > 0 
          ?  <>
              <p>Calorias totales consumidas: {calculateCurrentCaloriesOfGoal(props.goal)}/{props.goal.totalCalories}:</p> 
              <ProgressBar animated 
                variant={(calculateCurrentCaloriesOfGoal(props.goal)/props.goal.totalCalories)*100 > 65 ? "danger" : "success"}  
                now={(calculateCurrentCaloriesOfGoal(props.goal)/props.goal.totalCalories)*100} 
                label={`${(calculateCurrentCaloriesOfGoal(props.goal)/props.goal.totalCalories)*100}%`}
              />
            </>
          : null
          }
        </Modal.Body>
        <Modal.Footer>
          { isGoalEditable(props.goal)
            ? <Button type="button" onClick={() =>  {props.onUpdateCurrentGoalInState(props.goal); props.history.push("/goals", {goal: props.goal})}} style={{backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}}>Editar</Button>
            : null
          }
          <Button type="button" onClick={() =>  props.onDeleteGoal(props.goal.goalId, props.goal.userId)} style={{backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}}>Eliminar</Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}
  
  export default GoalModal;