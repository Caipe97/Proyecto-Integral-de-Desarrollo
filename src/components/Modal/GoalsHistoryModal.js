import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button } from 'react-bootstrap';
import GoalsSearchBar from '../SearchBar/GoalsSearchBar';

const GoalsHistoryModal =(props) =>{
  const [modalShow, setModalShow] = useState(false);
  return (
    <>
      <Button  type="button" variant="primary" style={{height: 50, width: "100%",backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}} onClick={() => setModalShow(true)}>
        Ver todas las metas
      </Button>
      
      <Modal
        show={modalShow}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
            {'Todas las Metas'}
          </Modal.Title>
          <Button type="button" onClick={() => setModalShow(false)} style={{ backgroundColor: 'white', borderColor: 'white', color: "black"}}>X</Button>
        </Modal.Header>
        <Modal.Body>
          <GoalsSearchBar {...props}/>
        </Modal.Body>
      </Modal>
    </>
  );
}
  
  export default GoalsHistoryModal
;