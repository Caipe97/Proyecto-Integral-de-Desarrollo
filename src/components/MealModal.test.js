import React from 'react';
import { shallow } from 'enzyme'; 
import MealModal from './MealModal';
import { MyVerticallyCenteredModal } from './MealModal';
import Modal from 'react-bootstrap/Modal';

describe('Modal tests', () => {
    const historyMock = { push: jest.fn() };///
    const mockProps = {mealId: 1,name:'comida1',dateEaten:'2021-09-09T00:00:00.000Z',FoodList:[]};
    const mealUpdate = {mealId: 1,name:'comida2',dateEaten:'2021-09-09T00:00:00.000Z',FoodList:[]};
//const preventDefault =  jest.fn();
    //const mockProps2= { show:false,onHide:() => {},meal:mockProps}
    let wrapper;
    const Delete= {mealId:1, userId:1}
    //let wrapper2;
    beforeEach(() => {
        wrapper = shallow(<MealModal meal={mockProps} onUpdateCurrentMealInState={mealUpdate} onDeleteMeal={Delete} history={historyMock}/>
            
        );

    })

    it('expect t2', () => {
        expect(wrapper).toMatchSnapshot();
    
    })
    it('expect to change  onclick buttons', () => {
       
        wrapper.find('[type="button"]').at(0).simulate('click');
        wrapper.find('[type="button"]').at(1).simulate('click');
        expect(wrapper.find('div')).toBeDefined();
        expect(wrapper.find('div.p').contains('')).toBeDefined();
        //wrapper.find('[type="button"]').at(3).simulate('click');
    })
//     it('expect to change update meals', async () => {
//     await wrapper.find('[type="button"]').at(2).simulate('click', preventDefault);
//    // expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(1);
//     //expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(1);
//     expect(historyMock.push.mock.calls[0]).toEqual(['/meals']);
//     })
})
