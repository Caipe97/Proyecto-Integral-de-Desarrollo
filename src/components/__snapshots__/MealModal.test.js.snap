// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests expect t2 1`] = `
<Fragment>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Mas info
  </Button>
  <Modal
    animation={true}
    aria-labelledby="contained-modal-title-vcenter"
    autoFocus={true}
    backdrop={true}
    centered={true}
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    keyboard={true}
    onDeleteMeal={
      Object {
        "mealId": 21,
        "userId": 1,
      }
    }
    onUpdateCurrentMealInState={
      Object {
        "FoodList": Array [
          Object {
            "food": Object {
              "caloriesPerServing": 198,
              "createdAt": "2021-09-15T19:58:04.486Z",
              "foodId": 1,
              "name": "Milanesa",
              "recommendedServing": 85,
            },
            "quantity": 1,
          },
        ],
      }
    }
    restoreFocus={true}
    show={false}
    size="lg"
  >
    <ModalHeader
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        id="contained-modal-title-vcenter"
      >
        comida1
      </ModalTitle>
    </ModalHeader>
    <ModalBody>
      <h4>
        2021-09-09 00:00
      </h4>
      <p>
        Alimentos consumidos:
      </p>
      <p>
        Total de calorias consumidas en la comida: 
        0
      </p>
    </ModalBody>
    <ModalFooter>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Cerrar
      </Button>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Editar
      </Button>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Eliminar
      </Button>
    </ModalFooter>
  </Modal>
</Fragment>
`;
